### PR TITLE
Meta: Cache VcPkg in CI, and move enablement of vcpkg to ladybird.sh and CMakePresets.json

### PR DESCRIPTION
**Pull request by ADKaster**
Also add a vcpkg command to ladybird.sh to ensure that vcpkg is setup,
and use a local binary cache for vcpkg build and install media to
avoid cluttering $XDG_CACHE_HOME.

Ref:

Binary caching in general https://learn.microsoft.com/en-us/vcpkg/users/binarycaching#default-binary-cache
GA cache: https://learn.microsoft.com/en-us/vcpkg/consume/binary-caching-github-actions-cache
Local cache: https://learn.microsoft.com/en-us/vcpkg/consume/binary-caching-local?pivots=shell-bash

### CODE REVIEW

**Comment by ADKaster** [comment]
cc @fgaz

Moving our vcpkg gunk out of the main CMakeLists.txt should allow for standard `find_package` or `find_package_handle_standard_args/pkgconfig` rules to apply for the dependencies we plan on adding, I hope?

The WIP woff2 dependency PR at least should work with a system package or a vcpkg package in my testing.

**Comment by trflynn89** [comment]
I'm not totally convinced it's an improvement to need to specify the vcpkg config in...6? places rather than just once in the root CMakeLists.txt. Especially given that our priority is development, not distribution.

**Comment by ADKaster** [comment]
If you ignore CI, it's only two places :)

It definitely feels cleaner to me... Setting a CMakeToolchain file is supposed to always be a command line option rather than a setting you put in your CMakeLists.txt. There's a reason I did it that way for the Serenity superbuild.

To me it clearly separates the role of vcpkg as a package manager from the build system. One could use any package manager, we've just settled on using vcpkg to make development easier.

**Comment by trflynn89** [comment]
With the toolchain set in CMakePresets.json, could we just use `cmake --preset=default` in CI and ladybird.sh? assuming all the extra definitions we add apply on top of that

**Comment by ADKaster** [comment]
Yes, though we would likely want to create a `CI` preset to add the extra options.

**Comment by trflynn89** [comment]
I tested with `cmake --preset=default -DENABLE_ADDRESS_SANITIZER=ON -B Build2` and it seems to work - no warnings or anything about it either

File: .github/workflows/lagom-template.yml [hunk 1]
```diff
@@ -59,6 +60,13 @@ jobs:
           serenity_ccache_path: ${{ env.CCACHE_DIR }}
           download_cache_path: ${{ github.workspace }}/Build/caches
 
+      - name: Export VcPkg GitHub Actions cache environment variables
```
Comment (trflynn89) [inline .github/workflows/lagom-template.yml:63]: not a big deal but I think the official casing is just "vcpkg", I don't see VcPkg used anywhere by microsoft

Comment (trflynn89) [reply .github/workflows/lagom-template.yml:63]: actual review comment: Does it make sense for this step to go into caches.yml?

**Review by trflynn89 — COMMENTED** [review]
Cool! Looks like a nice improvement to me, just a couple nits

**Review by trflynn89 — APPROVED** [review]

**Pull request merged**